src/clr-angular/accordion/accordion-panel.html

```diff
@@ -5,18 +5,17 @@
   </div>
 
   <div role="group" [ngClass]="getPanelStateClasses(panel)">
-    <div class="clr-accordion-header" [class.focus-within]="focusHeader">
+    <div class="clr-accordion-header">
       <button
         type="button"
         class="clr-accordion-header-button"
         (click)="togglePanel()"
-        (focus)="focusHeader = true"
-        (blur)="focusHeader = false"
         [id]="getAccordionHeaderId(panel.templateId)"
         [disabled]="panel.disabled"
         [attr.aria-controls]="getAccordionContentId(panel.templateId)"
         [attr.aria-expanded]="panel.open"
         [attr.aria-describedby]="getAccordionStatusId(panel.templateId)"
+        [class.clr-accordion-header-has-description]="description.children.length != 0"
```

Comment:
Regarding `description.children.length`,  we should avoid this DOM property traversal. Two reasons for this:
1. if the description gets removed for some reasons, the class binding wouldn't be updated as we haven't queried children DOM in Angular way.
2. As for server-side rendering, we should avoid traversing nested DOM properties as much as possible.

We could accomplish it more Angular way if we use `ContentChild` on `ClrAccordionTitle` or/and `ClrAccordionTitle`.

Reply:
Used `ContentChildren` to get updates about changes as using just `ContentChild` did not update if the description was dynamically added/remove.